File: meeting-rooms-iii.java
class Solution {
    public int mostBooked(int n, int[][] meetings) {
    PriorityQueue<Long> availRooms = new PriorityQueue<Long>();
    PriorityQueue<Long[]> assignQ = new PriorityQueue<Long[]>((a, b)-> Long.compare(a[1], b[1]));
    int roomMeetingCount[] = new int[n];
        
    for(int i=0; i<n; i++)
    availRooms.add((long)i);
    Arrays.sort(meetings, (a, b)-> a[0]-b[0]);
            
    long currTime = 0;
        
    for(int i=0; i<meetings.length; i++)
    {
      currTime = (long)Math.max(meetings[i][0], currTime);
        
      if(availRooms.isEmpty())
      {
        currTime = (long)Math.max(currTime, assignQ.peek()[1]);  
      }
        
      while(!assignQ.isEmpty() && currTime>=assignQ.peek()[1])
      {
        availRooms.add((long)assignQ.peek()[2]);
        assignQ.poll();  
      }
         long roomId = availRooms.poll(); 
         roomMeetingCount[(int)roomId] += 1;  
         assignQ.offer(new Long[]{currTime, currTime+(long)(meetings[i][1]-meetings[i][0]), roomId});
                   
    }
        
    int max = -1 ;
    int index = 0;     
    for(int p=0; p<n; p++) 
    {
      if(max<roomMeetingCount[p]) 
      {
          max=roomMeetingCount[p];
          index = p ;
      }
    }
        return index ; 
        
    }
}

File: meeting-rooms-iii.py
class Solution(object):
    def mostBooked(self, n, meetings):
        """
        :type n: int
        :type meetings: List[List[int]]
        :rtype: int
        """

        meetings.sort( key = lambda x: x[0])
        rooms=[i for i in range(n)]
        freq=[0]*n
        meets=[]
        for i in meetings:
            while meets and meets[0][0]<=i[0]:
                end,room = heappop(meets)
                heappush(rooms,room)
            if rooms:
                room=heappop(rooms)
                heappush(meets,(i[1],room))
            else:
                end,room = heappop(meets)
                heappush(meets,(end+(i[1]-i[0]),room))
            freq[room]+=1
        m=max(freq)
        print(freq)
        for i,j in enumerate(freq):
            if j==m:
                return i
